### ISSUE TITLE
Cannot create coffee and sql files

### ISSUE THREAD

**Issue by nguyenchr** [db-migrate/node-db-migrate#290]
Just wondering how the `node_module/.bin/db-migrate` file get’s created
I have npm install’ed the latest version and it seems that file is missing some functionality that seems to be present in `node-db-migrate/api.js`
to be specific this line is not there

``` js
 if (shouldCreateSqlFiles() && shouldCreateCoffeeFile()) {
      templateType = Migration.TemplateType.COFFEE_SQL_FILE_LOADER;
```
##

**Comment by wzrdtales** [comment]
You might confuse the actual released version and the current work in progress version.

This is the release branch of the actual published version:

https://github.com/db-migrate/node-db-migrate/tree/v0.9.x

The master contains everything related to the new version and what is currently work in progress and thus doesn't ment to be 100% production ready.

**Comment by nguyenchr** [comment]
I see, is there any plan to do a new release with all the changes in master soon?

**Comment by wzrdtales** [comment]
Yes, there are plans to do this soon.
I currently work on the linking between the new Seeders and Migrations, that is one of the last things I have to finish.

You can also look up the planned features, by viewing the milestone:

https://github.com/db-migrate/node-db-migrate/milestones/v0.10.0

You may notice, that some issues are the same feature. So generally there are really only a few tasks left, that is the linking, testing and verifying of the functionality of the seeders and finishing the programable API details. Well ok, I'm not ready to release v0.10 by this point. We're still missing tests and preparation work.

However, the plan is like the following:

Finishing the linking and programable API Details and publishing the first pre release (alpha) of v0.10. The next step is verifying the Seeder functionality, when this is verifed I will release the second pre release (beta). Next steps are providing all missing tests.

To give a list:

* alpha
Already a usable version, but Seeders might only be used with care.
* beta
A generally stable version, Seeders should work as expected, but aren't covered by tests yet.
* rc1
Providing tests to cover new programable API functionality.
* rc2
Providing tests to cover new Seeder functionality.
* final release
Polishing everything to release the final version.


RC1 and RC2 might be skipped completely, or only RC1 will be released before releasing the final v0.10.x. That depends and I can't say it yet.

I'm sorry that I can provide this information this late. But I had a tough time before, and I used the last weeks to catch up on all my projects I'm involved, including `db-migrate`. I'm now finally on track again, on all projects :)

For the changes you're looking for, you should be able to use the coming alpha release without thinking about stability. The alhpa state is more pointing to the actual Seeder state. All other functionality has indeed also changed a lot, but this functionality is already covered by tests and I ensured no change made any backward compatibility issues. If you just compare the v0.9.x branch with the current master, you see that I restructered a whole lot of thing and this was an important step for the future of db-migrate.

**Comment by nguyenchr** [comment]
thanks for the very thorough answer @wzrdtales :+1:
